src/lib.rs

```diff
@@ -1,9 +1,28 @@
+#![no_std]
+
+#[doc(hidden)]
+pub use core as _core;
+
 #[macro_export]
 macro_rules! opaque {
     ($name:ident) => {
         #[repr(C)]
         pub struct $name {
-            _private: [*mut u8; 0]
+            // Required for FFI-safe 0-sized type.
+            //
+            // In the future, this should refer to an extern type.
+            // See https://github.com/rust-lang/rust/issues/43467.
+            _private: [u8; 0],
```

Comment:
I'm wondering if that could not be removed now that the Phantom field is there?

Reply:
Unfortunately not. I get this error in the test:

> `extern` block uses type `test_t`, which is not FFI-safe composed only of `PhantomData`

Reply:
Hah! Which plays to #2.